sklearn/linear_model/logistic.py

```diff
@@ -1203,7 +1205,12 @@ def fit(self, X, y, sample_weight=None):
             raise ValueError("Tolerance for stopping criteria must be "
                              "positive; got (tol=%r)" % self.tol)
 
-        X, y = check_X_y(X, y, accept_sparse='csr', dtype=np.float64,
+        if self.solver in ['newton-cg']:
+            _dtype = [np.float64, np.float32]
```

Comment:
I think that @raghavrv was more concerned in the fact that we were passing a list rather than forcing one or the other. Once we checked that  `check_X_y` was taking care of it, he was ok with it.

@raghavrv any comments?